docs/build/guides/conventions/workspace.mdx

```diff
@@ -0,0 +1,357 @@
+---
+title: Workspaces
+hide_table_of_contents: true
+description: Organize contracts using Cargo workspaces
+---
+
+### Initializing workspace project
+
+Using Cargo's workspace [feature](https://doc.rust-lang.org/book/ch14-03-cargo-workspaces.html) makes it very convenient to organize your smart contracts in subdirectories of your project's root.
+
+It's very simple to get started using the cli:
+
+```
+stellar contract init . --name add_contract
+```
+
+Running this command will create a root project directory (`soroban-project`) and then initialize a project workspace with a single contract named `add_contract`.
+
+Adding one more contract template to the project can be done using the same command:
+
+```
+stellar contract init . --name main_contract
+```
+
+The project tree will look like this:
+
+```
+$ tree
+.
+├── Cargo.toml
+├── contracts
+│   ├── add-contract
+│   │   ├── Cargo.toml
+│   │   ├── Makefile
+│   │   └── src
+│   │       ├── lib.rs
+│   │       └── test.rs
+│   └── main-contract
+│       ├── Cargo.toml
+│       ├── Makefile
+│       └── src
+│           ├── lib.rs
+│           └── test.rs
+└── README.md
+```
+
+Running `stellar contract init` command created a sample contracts subdirectories (located in `./contracts), each contains:
+
+1. Cargo.toml file with all (minimal) necessary dependencies
+2. `Makefile` to easily build contract using `make` command
+3. `src` directory with an example hello-world contract and a simple test.
+
+We can now build our workspace's contracts with
+
+```
+stellar contract build
+```
+
+command and check the build directory `target/wasm32-unknown-unknown/release/`
+
+### Cross-contract call between workspaces
+
+With the given project structure, cross-contract calls can be easily made. Starting with modifying the sample hello world contract into an add contract:
+
+```rust title="contracts/add_contract/src/lib.rs"
+#![no_std]
+use soroban_sdk::{contract, contractimpl};
+
+#[contract]
+pub struct ContractAdd;
+
+#[contractimpl]
+impl ContractAdd {
+    pub fn add(x: u32, y: u32) -> u32 {
+        x.checked_add(y).expect("no overflow")
+    }
+}
+```
+
+The `ContractAdd` can now be referenced and used from another contracts in the same workspace using [`contractimport`](https://docs.rs/soroban-sdk/latest/soroban_sdk/macro.contractimport.html) macros. Note that the `contractimport` requires relative path of the compiled wasm file, so first it's required to recompile the contract with:
+
+```
+stellar contract build
+```
+
+As seen previously, target wasm file is located in
+
+```
+$PROJECT_ROOT/target/wasm32-unknown-unknown/release/
```

Comment:
We can assume the user is aware of the project root, and avoid the use of the env var they may not have set up:
```suggestion
target/wasm32-unknown-unknown/release/
```

Reply:
Deleted this section